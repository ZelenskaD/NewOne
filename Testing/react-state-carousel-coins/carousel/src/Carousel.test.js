import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";



//smoke test

it("renders without crashing", () => {
  render(<Carousel photos="photos" title="Shells from far-away beaches"/>);
});

//Snapshot test
it("renders without crashing", () => {
  const {asFragment} = render(<Carousel photos="photos" title="Shells from far-away beaches"/>);
  expect(asFragment()).toMatchSnapshot();
});




it("works when you click on the right arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});

//catches bug
it("works when you click on the left arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move to the left in the carousel
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  fireEvent.click(leftArrow);

  // expect the third image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 3"]')
  ).toBeInTheDocument();
});


test("left arrow is missing on the first image", () => {
  const { queryByTestId } = render(<Carousel photos={TEST_IMAGES} title="Test Carousel" />);

  // Check that the left arrow is not in the document
  const leftArrow = queryByTestId("left-arrow");
  expect(leftArrow).not.toBeInTheDocument();
});

test("right arrow is missing on the last image", () => {
  const { queryByTestId, getByTestId } = render(<Carousel photos={TEST_IMAGES} title="Test Carousel" />);

  // Move to the last image
  const rightArrow = getByTestId("right-arrow");
  fireEvent.click(rightArrow); // move to second image
  fireEvent.click(rightArrow); // move to third image

  // Check that the right arrow is not in the document
  const newRightArrow = queryByTestId("right-arrow");
  expect(newRightArrow).not.toBeInTheDocument();
});