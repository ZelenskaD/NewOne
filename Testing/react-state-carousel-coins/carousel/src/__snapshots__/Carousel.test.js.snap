// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far-away beaches
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        />
        <img
          class="Card-image"
        />
        <small
          class="Card-small"
        >
          Image 1 of 6.
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</DocumentFragment>
`;
